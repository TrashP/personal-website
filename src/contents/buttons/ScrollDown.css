@import '../../colors.css';

button.scroll {
display: block;
background-color: var(--color-light-bg);
border: solid 1px black;
border-radius: 50%;
width: 100px;
height: 100px;
margin-bottom: 5rem;
color: var(--color-dark-bg);
transition-property: border, color, box-shadow;
transition-duration: 0.2s;
transition-timing-function: ease-out;
}

button.scroll:hover {
    border: solid 2px var(--color-dark-secondary);
    color: var(--color-light-secondary);
    cursor: pointer;
    box-shadow: rgba(0, 0, 0, 0.16) 0px 10px 36px 0px, rgba(0, 0, 0, 0.06) 0px 0px 0px 1px;
}