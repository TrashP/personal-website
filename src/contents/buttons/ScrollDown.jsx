/* eslint-disable react/prop-types */

// ref to sibling element is passed down as prop from common parent
const ScrollDown = function ({ scrollRef }) {
  const handleClick = () => {
    scrollRef.current?.scrollIntoView({ behavior: 'smooth' });
  };

  return (
    <button type="button" className="scroll" onClick={handleClick}>
      Scroll Down
    </button>
  );
};

export default ScrollDown;
