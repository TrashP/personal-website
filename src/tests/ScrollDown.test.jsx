import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import ScrollDown from '../contents/buttons/ScrollDown';

describe('Scroll down button', () => {
  it('Renders scroll down button', () => {
    const { container } = render(<ScrollDown />);
    expect(container).toMatchSnapshot();
  });
});
