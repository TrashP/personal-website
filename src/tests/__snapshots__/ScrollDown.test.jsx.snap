// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Scroll down button > Renders scroll down button 1`] = `
<div>
  <button
    class="scroll"
    data-testid="id"
    type="button"
  >
    Scroll Down
  </button>
</div>
`;
